Explorer/Assets/Scripts/ECS/StreamableLoading/GLTF/LoadGLTFSystem.cs

```diff
@@ -69,8 +73,11 @@ protected override async UniTask<StreamableLoadingResult<GLTFData>> FlowInternal
                     new ReportData(GetReportCategory()),
                     new Exception("The content to download couldn't be found"));
 
-            gltfDownloadProvider.TargetGltfOriginalPath = intention.Name!;
-            var gltfImport = new GltfImport(downloadProvider: gltfDownloadProvider, logger: gltfConsoleLogger);
+            GltFastDownloadProvider gltfDownloadProvider = new GltFastDownloadProvider(World, sceneData, partitionComponent, intention.Name!);
```

Comment:
Is it necessary to create a new one for every GLTF download? Can it be resued?
`World`, `sceneData` and `partitionComponent` are constant (although I do think that you are not using `partitionComponent` correctly.
For what I see, `intention.Name!` is only used for debugging the exception. Can it be obtained or assigned in another way?

Reply:
Keep in mind raw GLTFs are only loaded for local scene development, not on prod.

Originally we tried reusing the `GLTFastDownloadProvider` instance but it's not easilly doable because of all the async approach... the same is done in unity-renderer, a new instance is created for every GLTF loaded, you can check the logic around [here](https://github.com/decentraland/unity-renderer/blob/dev/unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/GLTFast/AssetPromise_GLTFast_Loader.cs#L49) in case you want to confirm.

This is the problem that was happening until we stopped caching and sharing it:
1. GLTF "A" loading is started async [here](https://github.com/decentraland/unity-explorer/blob/3eccdb0898fe8b503504be80658330084ebf7f71/Explorer/Assets/Scripts/ECS/StreamableLoading/GLTF/LoadGLTFSystem.cs#L76). As you can see the 4th argument is the path of the GLTF (`intention.Name` is the path). That is used so that internally both `RequestAsync()` and `RequestTextureAsync()` look for the external dependencies in the same directory that the GLTF is in.
2. WHILE the GLTF "A" is still loading, GLTF "B" loading starts, thus modifying the property that holds the target GLTF original path/directory
3. WHILE Both GLTFs are still loading, an external texture for GLTF "A" is fetched, but now it fails because the GLTF original path/directory stores the value for GLTF "B" path, breaking the external dependencies fetching of GLTF "A".

As a side note, at least we are caching and using [only 1 instance of the MaterialGenerator](https://github.com/decentraland/unity-explorer/blob/3eccdb0898fe8b503504be80658330084ebf7f71/Explorer/Assets/Scripts/ECS/StreamableLoading/GLTF/LoadGLTFSystem.cs#L23).